includes/API/ApiLabkiUpdate.php

```diff
@@ -11,165 +11,336 @@
+            if ( $title && $title->exists() ) {
+                $wikiPage = WikiPage::factory( $title );
+                $wikiPage->doDeleteArticle( 'Removed via LabkiPackManager' );
+            }
+        }
+
+        // Remove from registry
+        $pageReg->removePagesByPack( $packId );
+        $packReg->removePack( $packId );
+
+        return [
+            'success' => true,
+            'action' => 'removePack',
+            'packId' => $packId,
+            'verify' => [
+                'providedCount' => is_array( $pages ) ? count( $pages ) : 0,
+                'removedCount' => is_array( $existingPages ) ? count( $existingPages ) : 0
+            ],
+            '_meta' => [ 'schemaVersion' => 1 ]
+        ];
+    }
+
+    /* -------------------------------------------------------------------------
+     * PAGE IMPORT HELPERS
+     * ------------------------------------------------------------------------- */
+    private function importPackPages( string $repoUrl, array $pack, array $rewriteMap ): array {
+        $pagesOut = [];
+        $pages = $pack['pages'] ?? [];
+
+        foreach ( $pages as $p ) {
+            $relPath = $p['path'] ?? null;
+            $finalTitle = $p['finalTitle'] ?? '';
+            $sourceName = $p['original'] ?? $p['name'] ?? '';
+            if ( $relPath === null || $finalTitle === '' ) {
+                continue;
+            }
+
+            $fileUrl = rtrim( $repoUrl, '/' ) . '/' . ltrim( $relPath, '/' );
+            $wikitext = $this->fetchWikiFile( $fileUrl );
+            if ( $wikitext === null ) {
+                wfDebugLog( 'Labki', "Failed to fetch wiki file: $fileUrl" );
+                continue;
+            }
+
+            // Rewrite internal links
+            $updatedText = $this->rewriteLinks( $wikitext, $rewriteMap );
+
+            $title = Title::newFromText( $finalTitle );
+            if ( !$title ) {
+                continue;
+            }
+
+            $content = new WikitextContent( $updatedText );
+            $page = WikiPage::factory( $title );
+            $editResult = $page->doEditContent( $content, 'Imported via LabkiPackManager' );
+
+            if ( !$editResult->isOK() ) {
+                wfDebugLog( 'Labki', "Edit failed for $finalTitle" );
+                continue;
+            }
+
+            $pagesOut[] = [
+                'name' => $sourceName,
+                'finalTitle' => $finalTitle,
+                'pageNamespace' => $title->getNamespace(),
+                'wikiPageId' => $page->getId()
+            ];
+        }
+
+        return $pagesOut;
+    }
+
+    private function fetchWikiFile( string $url ): ?string {
+        $ctx = stream_context_create( [ 'http' => [ 'timeout' => 10 ] ] );
+        $data = @file_get_contents( $url, false, $ctx );
+        return $data !== false ? $data : null;
+    }
+
+    private function rewriteLinks( string $text, array $map ): string {
```

Comment:
def need tests for this because a failure here, esp with templates, could propagate through the whole pack and break a lot of pages

Reply:
Yep. Will add tests soon